### PR TITLE
add runsettings file

### PR DESCRIPTION
**Pull request by juandent**
to allow Catch2 to find tests. It is also necessary to set in Visual Studio 2022 Test\Configure Run Settings as shown in this image:

image

### CODE REVIEW

**Comment by fnc12** [comment]
@juandent what problems it solves?

**Comment by juandent** [comment]
Without  the runsettings, the tests are not found... at least not in my machine. It could be you have a global runsettings and so you do not need it, I don't know. Please check

**Comment by fnc12** [comment]
@juandent I use Xcode or raw cmake on macOS. Is it something related to Visual Studio only?

**Comment by juandent** [comment]
could be 
image

**Comment by fnc12** [comment]
why Windows works well on CI machines without this file?

**Comment by juandent** [comment]
but additionally VS2022 must be made to point to it

**Comment by juandent** [comment]
who configured VS2022 in the CI mnachines?

**Comment by trueqbit** [comment]
sqlite_orm comes with CMake build scripts, which in turn uses the specified compiler. CMake doesn't use Visual Studio per se, and even if they are kind of tied together, Visual Studio is not the msvc compiler, but ships with the msvc compiler.

What you are referring to is when you open a "solution" in the Visual Studio IDE.

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,8 @@ if (MSVC)
         message(STATUS "Add /bigobj flag to compiler")
         add_compile_options(/bigobj)
     endif()
+    # For Visual Studio users
+    file(COPY_FILE ${PROJECT_SOURCE_DIR}/CMake/.runsettings ${PROJECT_BINARY_DIR}/.runsettings )
```
Comment (trueqbit) [inline CMakeLists.txt:78]: Need to conditionally copy the file: `if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.21)`

File: cmake/.runsettings
Comment (trueqbit) [inline cmake/.runsettings:1]: I'd choose a better name for the file, e.g. _vs.runsettings_.

**Review by trueqbit — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (juandent) [reply CMakeLists.txt]: Ok I conditioned the call

**Pull request closed without merging**